--- a/ext/mpeg2dec/gstmpeg2dec.c
+++ b/ext/mpeg2dec/gstmpeg2dec.c
@@ -488,7 +488,7 @@ gst_mpeg2dec_crop_buffer (GstMpeg2dec * dec, GstVideoCodecFrame * in_frame,
 {
   GstVideoCodecState *state;
   GstVideoInfo *info;
-  GstVideoInfo *dinfo;
+  GstVideoInfo *dinfo G_GNUC_UNUSED;
   GstVideoFrame output_frame;
   GstFlowReturn ret;
   GstBuffer *buffer = NULL;
@@ -878,7 +878,7 @@ handle_picture (GstMpeg2dec * mpeg2dec, const mpeg2_info_t * info,
   GstVideoDecoder *decoder = (GstVideoDecoder *) mpeg2dec;
   GstFlowReturn ret;
   gint type;
-  const gchar *type_str = NULL;
+  const gchar *type_str G_GNUC_UNUSED = NULL;
   gboolean key_frame = FALSE;
   const mpeg2_picture_t *picture = info->current_picture;
   GstVideoFrame vframe;
diff --git a/ext/x264/gstx264enc.c b/ext/x264/gstx264enc.c
index 7d5e2b4..74ea215 100644
--- a/ext/x264/gstx264enc.c
+++ b/ext/x264/gstx264enc.c
@@ -1667,7 +1667,7 @@ gst_x264_enc_header_buf (GstX264Enc * encoder)
   int nal_size;
   guint8 *buffer, *sps;
   gulong buffer_size;
-  gint sei_ni = 2, sps_ni = 0, pps_ni = 1;
+  gint sei_ni G_GNUC_UNUSED = 2, sps_ni = 0, pps_ni = 1;
 
   if (G_UNLIKELY (encoder->x264enc == NULL))
     return NULL;
diff --git a/gst/asfdemux/gstasfdemux.c b/gst/asfdemux/gstasfdemux.c
index 1205a6f..3f60063 100755
--- a/gst/asfdemux/gstasfdemux.c
+++ b/gst/asfdemux/gstasfdemux.c
@@ -2727,7 +2727,7 @@ gst_asf_demux_parse_stream_object (GstASFDemux * demux, guint8 * data,
 {
   AsfCorrectionType correction_type;
   AsfStreamType stream_type;
-  GstClockTime time_offset;
+  GstClockTime time_offset G_GNUC_UNUSED;
   gboolean is_encrypted G_GNUC_UNUSED;
   guint16 stream_id;
   guint16 flags;
@@ -2795,7 +2795,8 @@ gst_asf_demux_parse_stream_object (GstASFDemux * demux, guint8 * data,
 
       switch (correction_type) {
         case ASF_CORRECTION_ON:{
-          guint span, packet_size, chunk_size, data_size, silence_data;
+          guint span, packet_size, chunk_size, data_size,
+              silence_data G_GNUC_UNUSED;
 
           GST_INFO ("Using error correction");
 
@@ -2860,7 +2861,7 @@ gst_asf_demux_parse_stream_object (GstASFDemux * demux, guint8 * data,
     case ASF_STREAM_VIDEO:{
       asf_stream_video_format video_format_object;
       asf_stream_video video_object;
-      guint16 vsize;
+      guint16 vsize G_GNUC_UNUSED;
 
       if (!gst_asf_demux_get_stream_video (&video_object, &data, &size))
         goto not_enough_data;
@@ -3415,7 +3416,7 @@ gst_asf_demux_process_file (GstASFDemux * demux, guint8 * data, guint64 size)
   guint64 creation_time G_GNUC_UNUSED;
   guint64 file_size G_GNUC_UNUSED;
   guint64 send_time G_GNUC_UNUSED;
-  guint64 packets_count, play_time, preroll;
+  guint64 packets_count G_GNUC_UNUSED, play_time, preroll;
   guint32 flags, min_pktsize, max_pktsize, min_bitrate G_GNUC_UNUSED;
 
   if (size < (16 + 8 + 8 + 8 + 8 + 8 + 8 + 4 + 4 + 4 + 4))
@@ -3824,8 +3825,8 @@ gst_asf_demux_process_ext_stream_props (GstASFDemux * demux, guint8 * data,
   guint16 num_payload_ext;
   guint64 len;
   guint8 *stream_obj_data = NULL;
-  guint8 *data_start;
-  guint obj_size;
+  guint8 *data_start G_GNUC_UNUSED;
+  guint obj_size G_GNUC_UNUSED;
   guint i, stream_num;
 
   data_start = data;
diff --git a/gst/asfdemux/gstrtspwms.c b/gst/asfdemux/gstrtspwms.c
index c864287..47398c6 100644
--- a/gst/asfdemux/gstrtspwms.c
+++ b/gst/asfdemux/gstrtspwms.c
@@ -129,7 +129,7 @@ no_config:
 static gboolean
 gst_rtsp_wms_configure_stream (GstRTSPExtension * ext, GstCaps * caps)
 {
-  GstRTSPWMS *ctx;
+  GstRTSPWMS *ctx G_GNUC_UNUSED;
   GstStructure *s;
   const gchar *encoding;
 
@@ -153,7 +153,7 @@ gst_rtsp_wms_configure_stream (GstRTSPExtension * ext, GstCaps * caps)
 static GstRTSPResult
 gst_rtsp_wms_receive_request (GstRTSPExtension * ext, GstRTSPMessage * request)
 {
-  GstRTSPWMS *ctx;
+ GstRTSPWMS *ctx G_GNUC_UNUSED;
   GstRTSPResult res = GST_RTSP_ENOTIMPL;
   GstRTSPMessage response = { 0 };
 
diff --git a/gst/realmedia/rdtmanager.c b/gst/realmedia/rdtmanager.c
index 39c74b2..871561a 100644
--- a/gst/realmedia/rdtmanager.c
+++ b/gst/realmedia/rdtmanager.c
@@ -591,7 +591,7 @@ gst_rdt_manager_src_activate_mode (GstPad * pad, GstObject * parent,
     GstPadMode mode, gboolean active)
 {
   gboolean result;
-  GstRDTManager *rdtmanager;
+  GstRDTManager *rdtmanager G_GNUC_UNUSED;
   GstRDTManagerSession *session;
 
   session = gst_pad_get_element_private (pad);
@@ -648,8 +648,8 @@ static GstFlowReturn
 gst_rdt_manager_handle_data_packet (GstRDTManagerSession * session,
     GstClockTime timestamp, GstRDTPacket * packet)
 {
-  GstRDTManager *rdtmanager;
-  guint16 seqnum;
+  GstRDTManager *rdtmanager G_GNUC_UNUSED;
+  guint16 seqnum G_GNUC_UNUSED;
   gboolean tail;
   GstFlowReturn res;
   GstBuffer *buffer;
@@ -856,7 +856,7 @@ gst_rdt_manager_chain_rdt (GstPad * pad, GstObject * parent, GstBuffer * buffer)
 static void
 gst_rdt_manager_loop (GstPad * pad)
 {
-  GstRDTManager *rdtmanager;
+  GstRDTManager *rdtmanager G_GNUC_UNUSED;
   GstRDTManagerSession *session;
   GstBuffer *buffer;
   GstFlowReturn result;
@@ -938,7 +938,7 @@ static GstFlowReturn
 gst_rdt_manager_chain_rtcp (GstPad * pad, GstObject * parent,
     GstBuffer * buffer)
 {
-  GstRDTManager *src;
+  GstRDTManager *src G_GNUC_UNUSED;
 
 #ifdef HAVE_RTCP
   gboolean valid;
diff --git a/gst/realmedia/rmdemux.c b/gst/realmedia/rmdemux.c
index d5d3ac1..e79bc81 100644
--- a/gst/realmedia/rmdemux.c
+++ b/gst/realmedia/rmdemux.c
@@ -2215,7 +2215,7 @@ gst_rmdemux_parse_video_packet (GstRMDemux * rmdemux, GstRMDemuxStream * stream,
     guint8 pkg_header;
     guint pkg_offset;
     guint pkg_length;
-    guint pkg_subseq = 0, pkg_seqnum = G_MAXUINT;
+    guint pkg_subseq = 0, pkg_seqnum G_GNUC_UNUSED = G_MAXUINT;
     guint fragment_size;
     GstBuffer *fragment;
 
