--- a/ext/libav/gstavdemux.c
+++ b/ext/libav/gstavdemux.c
@@ -981,7 +981,7 @@ gst_ffmpegdemux_get_stream (GstFFMpegDemux * demux, AVStream * avstream)
 
   /* transform some useful info to GstClockTime and remember */
   {
-    GstClockTime tmp;
+    GstClockTime tmp G_GNUC_UNUSED;
 
     /* FIXME, actually use the start_time in some way */
     tmp = gst_ffmpeg_time_ff_to_gst (avstream->start_time, avstream->time_base);
diff --git a/ext/libav/gstavviddec.c b/ext/libav/gstavviddec.c
index 0521aec..ecc6505 100644
--- a/ext/libav/gstavviddec.c
+++ b/ext/libav/gstavviddec.c
@@ -1430,7 +1430,7 @@ gst_ffmpegviddec_frame (GstFFMpegVidDec * ffmpegdec,
     guint8 * data, guint size, gint * have_data, GstVideoCodecFrame * frame,
     GstFlowReturn * ret)
 {
-  GstFFMpegVidDecClass *oclass;
+  GstFFMpegVidDecClass *oclass G_GNUC_UNUSED;
   gint len = 0;
 
   if (G_UNLIKELY (ffmpegdec->context->codec == NULL))
diff --git a/ext/libav/gstavvidenc.c b/ext/libav/gstavvidenc.c
index 0e90dc7..1da4208 100644
--- a/ext/libav/gstavvidenc.c
+++ b/ext/libav/gstavvidenc.c
@@ -286,7 +286,7 @@ gst_ffmpegvidenc_finalize (GObject * object)
 static GstCaps *
 gst_ffmpegvidenc_getcaps (GstVideoEncoder * encoder, GstCaps * filter)
 {
-  GstFFMpegVidEnc *ffmpegenc = (GstFFMpegVidEnc *) encoder;
+  GstFFMpegVidEnc *ffmpegenc G_GNUC_UNUSED = (GstFFMpegVidEnc *) encoder;
   GstCaps *caps = NULL;
 
   GST_DEBUG_OBJECT (ffmpegenc, "getting caps");
